### PR TITLE
Functional gateway tests now use real DynamoDb instance

### PR DESCRIPTION
**Pull request by mikemoody-amido**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/MTTL-538

## Describe this PR

Update the Gateway class tests so that they use a real DynamoDb instance where possible.

### CODE REVIEW

File: PersonApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs [hunk 1]
```diff
@@ -13,19 +13,20 @@
...
         private readonly Fixture _fixture = new Fixture();
-        private readonly Mock<IDynamoDBContext> _dynamoDb;
         private readonly Mock<ILogger<DynamoDbGateway>> _logger;
-        private readonly DynamoDbGateway _classUnderTest;
+        private DynamoDbGateway _classUnderTest;
 
-        public DynamoDbGatewayTests()
+        private readonly IDynamoDBContext _dynamoDb;
+
+        public DynamoDbGatewayTests(DynamoDbIntegrationTests<Startup> dbTestFixture)
```
Comment (humulla) [inline PersonApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs:25]: What if you inject the integrationTests instead

Comment (humulla) [inline PersonApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs:25]: The 'integrationTests' file should be creating a table that can then be used when the SaveAsync() is called

**Review by humulla — COMMENTED** [review]
I would advise taking a look at the changes I'm making to my [gateway tests](https://github.com/LBHackney-IT/contact-details-api/tree/gateway-tests) so both our APIs are consistent and running in a similar format

**Comment by mikemoody-amido** [comment]
> I would advise taking a look at the changes I'm making to my [gateway tests](https://github.com/LBHackney-IT/contact-details-api/tree/gateway-tests) so both our APIs are consistent and running in a similar format

Your tests are still using nUnit, whereas these are using Xunit meaning the approaches are necessarily different.

**Comment by humulla** [comment]
> > I would advise taking a look at the changes I'm making to my [gateway tests](https://github.com/LBHackney-IT/contact-details-api/tree/gateway-tests) so both our APIs are consistent and running in a similar format
> 
> Your tests are still using nUnit, whereas these are using Xunit meaning the approaches are necessarily different.

I hear your point, but I guess my main question is how does your saveAsync() know which table to save the data in, as within the integrationTest the tabledef is only defined the actual table is not being created

**Comment by mikemoody-amido** [comment]
The table _will_ be created if it is not there already.

**Comment by humulla** [comment]
Ah okay, that makes sense, thank you for clarifying that for me :slightly_smiling_face:

**Review by humulla — APPROVED** [review]

File: PersonApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs [hunk 2]
```diff
@@ -44,37 +45,40 @@ public async Task GetPersonByIdReturnsNullIfEntityDoesntExist()
...
             // Act
             var response = await _classUnderTest.GetPersonByIdAsync(entity.Id).ConfigureAwait(false);
 
             // Assert
-            _dynamoDb.Verify(x => x.LoadAsync<PersonDbEntity>(dbIdUsed, default), Times.Once);
-            entity.Id.Should().Be(response.Id);
+            entity.Should().BeEquivalentTo(response);
             _logger.VerifyExact(LogLevel.Debug, $"Calling IDynamoDBContext.LoadAsync for id {entity.Id}", Times.Once());
+
+            await _dynamoDb.DeleteAsync(dbEntity).ConfigureAwait(false);
```
Comment (amidomichael) [inline PersonApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs:62]: Do we create/destroy a temporary table for this test? If, so, is this needed?

**Pull request merged**